### PR TITLE
Minor Requirement fixes

### PR DESCRIPTION
**Pull request by yjskrs**

### CODE REVIEW

File: src/main/java/igrad/logic/commands/requirement/RequirementAddCommand.java [hunk 1]
```diff
@@ -15,17 +15,20 @@
...
         + "Example: " + COMMAND_WORD + " "
         + PREFIX_NAME + "Unrestricted Electives "
         + PREFIX_CREDITS + "24\n";
 
+    public static final String MESSAGE_HELP = MESSAGE_DETAILS + MESSAGE_USAGE;
+
     public static final String MESSAGE_REQUIREMENT_ADD_SUCCESS = "New requirement added: %1$s";
     public static final String MESSAGE_REQUIREMENT_NOT_ADDED = "Added requirement must be provided with arguments "
-        + PREFIX_NAME + "TITLE " + PREFIX_CREDITS + "CREDITS ";
+        + PREFIX_NAME + "NAME " + PREFIX_CREDITS + "CREDITS ";
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/commands/requirement/RequirementAddCommand.java:31]: ok thanks for changing  it back to `NAME`, sorry i didn't changed it properly the  last time

File: src/main/java/igrad/logic/parser/Prefix.java [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 /**
  * A prefix that marks the beginning of an argument in an arguments string.
- * E.g. 't/' in 'add James t/ friend'.
+ * E.g. 'n/' in 'module add n/CS1101S t/Programming Methodology I'.
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/parser/Prefix.java:5]: ok nice

File: src/main/java/igrad/logic/parser/requirement/RequirementAddCommandParser.java [hunk 3]
```diff
@@ -11,29 +11,29 @@
...
 import igrad.logic.parser.exceptions.ParseException;
 import igrad.model.requirement.Credits;
 import igrad.model.requirement.Name;
 import igrad.model.requirement.Requirement;
 
 /**
- * Parses requirement input argument and creates a new RequirementAddCommand object.
+ * Parses requirement input arguments and returns a new RequirementAddCommand object.
  */
-public class RequirementAddCommandParser extends RequirementCommandParser implements Parser<RequirementAddCommand> {
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/parser/requirement/RequirementAddCommandParser.java:24]: oh nice, good catch

File: src/main/java/igrad/logic/parser/requirement/RequirementCommandParser.java [hunk 4]
```diff
@@ -2,41 +2,48 @@
...
+     * @throws ParseException If the given {@code name} is invalid.
      */
     public static Name parseName(String name) throws ParseException {
         requireNonNull(name);
+
         String trimmedName = name.trim();
-        if (!Name.isValidTitle(trimmedName)) {
-            throw new ParseException(igrad.model.module.Title.MESSAGE_CONSTRAINTS);
+        if (!Name.isValidName(trimmedName)) {
+            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/parser/requirement/RequirementCommandParser.java:27]: 👍

File: src/main/java/igrad/model/requirement/Credits.java [hunk 5]
```diff
@@ -5,24 +5,26 @@
...
+ * Guarantees: immutable, non-null and is valid as declared by {@link #isValidCredits(String)}
  */
 public class Credits {
 
-    public static final String MESSAGE_CONSTRAINTS = "Modular credits should contain one or more digits!";
+    public static final String MESSAGE_CONSTRAINTS =
+        "Credits needed to satisfy requirement should be more than 0.";
 
-    public static final String VALIDATION_REGEX = "\\d+";
+    public static final String VALIDATION_REGEX = "^[0-9]\\d*$"; // allows any numbers more than or equals zero
```
Comment (nathanaelseen) [inline src/main/java/igrad/model/requirement/Credits.java]: hmm why equals to zero though, no modules can be  added under it then when i start to work on assign haha

File: src/main/java/igrad/model/requirement/Name.java [hunk 6]
```diff
@@ -4,37 +4,33 @@
...
+        checkArgument(isValidName(value), MESSAGE_CONSTRAINTS);
+
+        this.value = value;
     }
 
     /**
-     * Returns true if a given string is a valid title.
+     * Returns true if a given string is a valid name.
      */
-    public static boolean isValidTitle(String test) {
```
Comment (nathanaelseen) [inline src/main/java/igrad/model/requirement/Name.java:37]: oh yeah ok sry once again for not changing it properly

File: src/main/java/igrad/model/requirement/exceptions/DuplicateRequirementException.java [hunk 7]
```diff
@@ -1,7 +1,7 @@
 package igrad.model.requirement.exceptions;
 
 /**
- * Signals that the operation will result in duplicate requirements with the same title.
```
Comment (nathanaelseen) [inline src/main/java/igrad/model/requirement/exceptions/DuplicateRequirementException.java:4]: this too

**Review by nathanaelseen — APPROVED** [review]

File: src/main/java/igrad/model/requirement/Credits.java (hunk 5 above)
Comment (nathanaelseen) [inline src/main/java/igrad/model/requirement/Credits.java:15]: the comments are abit misleading, at first i thought the regex allowed zero too

Comment (nathanaelseen) [reply src/main/java/igrad/model/requirement/Credits.java:15]: oh wait actually it allowed 0 too, suggestion; `^[1-9]\\d*$`

Comment (yjskrs) [reply src/main/java/igrad/model/requirement/Credits.java:15]: it allows zero.
i dont want to reject users who put 04 or 024 or something.

Comment (nathanaelseen) [reply src/main/java/igrad/model/requirement/Credits.java:15]: then i guess you've to think of some regex to prevent '0'

**Pull request merged**